PlaywrightSolar/PageObjects/SolarPanelsPage.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Playwright;
+
+namespace PlaywrightSolar.PageObjects;
+
+public class SolarPanelsPage(IPage page)
+{
+    private const string SolarPanelsPageUrl = "https://solartechnology.com.ua/shop/solar-panels";
+    
+    // Locators
+    private ILocator PageTitle => page.Locator("//h1");
+
+    private ILocator ProductFilterButton => page.GetByText("Фільтр товарів");
+
+    private readonly Dictionary<string, ILocator> _filterLocators = new()
+    {
+        { "Abi-Solar", page.GetByText("Abi-Solar", new PageGetByTextOptions { Exact = true }) },
+        { "C&T Solar", page.GetByText("C&T Solar") },
+        { "JA Solar", page.GetByText("JA Solar") },
+        { "Jinko Solar", page.GetByText("Jinko Solar") },
+        { "SOLA", page.GetByText("SOLA", new PageGetByTextOptions { Exact = true }) },
+        { "Ulica Solar", page.GetByText("Ulica Solar") },
+        { "Yingli Solar", page.GetByText("Yingli Solar") }
+    };
+    
+    private readonly Dictionary<string, ILocator> _panelTypeFiltersLocator = new()
+    {
+        { "Монокристал", page.GetByText("Монокристал", new PageGetByTextOptions { Exact = true }) },
+        { "Полікристал", page.GetByText("Полікристал", new PageGetByTextOptions { Exact = true }) }
+    };
+
+    // Methods
+    public async Task GoToSolarPanelsPage()
+    {
+        await page.GotoAsync(SolarPanelsPageUrl);
+        Assert.That(page.Url, Is.EqualTo(SolarPanelsPageUrl), "Failed to navigate to Solar Panels page");
+    }
+
+    public async Task SolarPanelsPageOpened()
+    {
+        await Assertions.Expect(PageTitle).ToBeVisibleAsync();
+    }
+
+    public async Task OpenProductFilter()
+    {
+        await ProductFilterButton.ClickAsync();
+        await Task.Delay(1000);
+    }
+    
+    public async Task<List<string>> GetInitialProductsName()
+    {
+        await page.WaitForSelectorAsync(".prod-holder");
+        var results = await page.QuerySelectorAllAsync(".prod-holder");
+        var resultText = await Task.WhenAll(results.Select(async result => await result.InnerTextAsync()));
+        return resultText.ToList();
+    }
+    
+    public async Task<List<string>> GetProductsNameByManufacturer(string manufacturer)
```

Comment:
name does not match to implementation. It does not GetProductsNameByManufacturer, it actually filters products by manufacture and then returns all product texts from the page after filtering. Make this method FilterByManufacturer and in the test you will have much clearer steps like
var initialProductNames = ...GetProductNames()
FilterByManufacturer 
var productNamesAfterFiltering = ...GetProductNames()
compare before and after